File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FCheckBox.java
package io.github.gofaith.faithdroid.FViews;

import android.graphics.Color;
import android.view.View;
import android.widget.CheckBox;
import android.widget.CompoundButton;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.UIController;

public class FCheckBox extends FView implements AttrGettable, AttrSettable {
    private CheckBox v;

    public FCheckBox(UIController controller) {
        parentController = controller;
        v = new CheckBox(parentController.activity);
        view=v;
    }
    @Override
    public String getAttr(String attr) {
        String str = getUniversalAttr(attr);
        if (str != null) {
            return str;
        }
        switch (attr){
            // ------------------------------------------
            case "Checked":
                return String.valueOf(v.isChecked());
            case "Enabled":
                return String.valueOf(v.isEnabled());
            case "Text":
                return v.getText().toString();
        }
        return "";
    }

    @Override
    public void setAttr(String attr, final String value) {
        if (setUniversalAttr(attr, value)) {
            return;
        }
        switch (attr){
            // -------------------------------------------------------------------
            case "Text":
                v.setText(value);
                break;
            case "TextColor":
                try {
                    v.setTextColor(Color.parseColor(value));
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "TextSize":
                try {
                    v.setTextSize(dp2pixel(parentController.activity,Float.valueOf(value)));
                } catch (Exception e) {
                    e.printStackTrace();
                    return;
                }
                break;
            case "Enabled":
                if (value.equals("true")) {
                    v.setEnabled(true);
                } else {
                    v.setEnabled(false);
                }
                break;
            case "OnCheckedChangeListener":
                v.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                    @Override
                    public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                        Faithdroid.triggerEventHandler(value, String.valueOf(isChecked));
                    }
                });
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FCheckBox.kt
package io.github.stevenzack.faithdroidapp.FViews

import android.graphics.Color
import android.widget.CheckBox
import android.widget.CompoundButton
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.AttrGettable
import io.github.stevenzack.faithdroidapp.UI.AttrSettable
import io.github.stevenzack.faithdroidapp.UI.FView
import io.github.stevenzack.faithdroidapp.UI.Toolkit.dp2pixel
import io.github.stevenzack.faithdroidapp.UI.UIController

class FCheckBox(controller: UIController) : FView(), AttrGettable, AttrSettable {
    private val v: CheckBox
    internal var onChange: String? = null

    init {
        parentController = controller
        v = CheckBox(parentController!!.activity)
        view = v
    }

  override  fun getAttr(attr: String): String {
        val str = getUniversalAttr(attr)
        if (str != null) {
            return str
        }
        when (attr) {
            // ------------------------------------------
            "Checked" -> return v.isChecked.toString()
            "Enabled" -> return v.isEnabled.toString()
            "Text" -> return v.text.toString()
        }
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (setUniversalAttr(attr, value)) {
            return
        }
        when (attr) {
            // -------------------------------------------------------------------
            "Text" -> v.text = value
            "TextColor" -> try {
                v.setTextColor(Color.parseColor(value))
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "TextSize" -> try {
                v.textSize = dp2pixel(parentController!!.activity, java.lang.Float.valueOf(value))
            } catch (e: Exception) {
                e.printStackTrace()
                return
            }

            "Enabled" -> if (value == "true") {
                v.isEnabled = true
            } else {
                v.isEnabled = false
            }
            "OnCheckedChangeListener" -> {
                if (onChange == null) {
                    v.setOnCheckedChangeListener { buttonView, isChecked ->
                        Faithdroid.triggerEventHandler(
                            onChange,
                            isChecked.toString()
                        )
                    }
                }
                onChange = value
            }
            "Checked" -> v.isChecked = value == "true"
        }
    }
}
